### DISCUSSION TITLE
Template switch e call service switch.turn_off

### DISCUSSION THREAD

**Discussion by Cionco65** [indomus/forum#1963] [HUB: Home Assistant]
Buongiorno a tutti.

Dopo anni di uso di home assistant mi è sorto un dubbio sui template switches.
Ho creato un template switch per comandare un riscaldatore tramite broadlink che prende lo stato da un sonoff pow (>5 watt = on, altrimenti off).
Il comando inviato al broadlink è sempre lo stesso (unico pulsante su telecomando), il fatto è che a seguito di una chiamata tramite servizio switch.turn_off il sistema sembra eseguire le azioni indicate sotto la direttiva "turn_off" del template non prendendo in considerazione lo stato attuale dello switch.
Il risultato è che se se il riscaldatore è spento, richiamando switch.turn_off, anzichè non fare nulla, il servizio me lo accende!!!

Il problema è che ho diversi template di questo tipo e sono stati messi in un gruppo per gestire il "fuori casa". 
Gestirli uno ad uno, seppur fattibile, comporterebbe una complicazione ulteriore.

E' un comportamento regolare?

Grazie, Paolo

**Comment by snakuzzo** [comment]
Posta il codice che hai usato, utilizzando il markdown

**Reply by 50494554524F** [reply]
ovvio che lo esegue se chiami il servizio... metti un controllo sullo stato dello switch all'inizio di turn_off

**Reply by Cionco65** [reply]
Beh, proprio ovvio non direi, poteva anche essere immaginabile un'astrazione maggiore dal template. La mia infatti era una richiesta di conferma.
Grazie per avermi risposto :)

per Diego, questo è lo yaml:

```
template:
  - binary_sensor:
      - name: argo_2000 # Sensore Argo a 2kw
        unique_id: argo_2000
        state: >
          {% if (states('sensor.sonoff_pow_argo_power')|int >= 1500) %}
          on
          {% else %}
          off
          {% endif %}  

      - name: argo_1000 # Sensore Argo a 1kw
        unique_id: argo_1000
        state: >
          {% if (states('sensor.sonoff_pow_argo_power')|int > 500 and states('sensor.sonoff_pow_argo_power')|int < 1500) %}
          on
          {% else %}
          off
          {% endif %}  
          
      - name: argo_status # Sensore Argo a acceso/spento
        unique_id: argo_status
        state: >
          {% if (states('sensor.sonoff_pow_argo_power')|int > 5) %}
          on
          {% else %}
          off
          {% endif %}  
switch:
  - platform: template
    switches:
      argo:
        friendly_name: 'Argo on-off'
        unique_id: argo_on_off
        value_template: "{{ is_state('binary_sensor.argo_status', 'on') }}"
        turn_on:
          service: remote.send_command
          target:
            entity_id: remote.rm_up_remote
          data:
            device: argo_chic
            command: onoff
        turn_off:
          service: remote.send_command
          target:
            entity_id: remote.rm_up_remote
          data:
            device: argo_chic
            command: onoff

**Reply by 50494554524F** [reply]
ti basta aggiungere questo all'inizio dei turn_on/turn_off

```
          turn_on:
            - condition: state
              entity_id: switch.nomeswitch
              state: 'off'

          turn_off:
            - condition: state
              entity_id: switch.nomeswitch
              state: 'on'
```
e poi il resto dello script...

**Reply by Cionco65** [reply]
Come immaginavo.
Ti ringrazio, modificherò i miei template in tal senso.
